file.go

```diff
@@ -1,37 +1,140 @@
 package jcsv
 
-import "os"
+import (
+	"encoding/csv"
+	"encoding/json"
+	"fmt"
+	"io/ioutil"
+	"os"
+	"strings"
+)
 
 type file struct {
-	// TODO: define variables if needed
+	data []map[string]interface{} //a 1D-array of maps having string keys and empty interface values
 }
 
 func ParseJsonFile(path string) (file, error) {
-	// TODO: open and read the given file into your `file` object
-	return file{}, nil
+	jsonFile, openError := os.Open(path)
+	if openError != nil {
+		return file{}, openError
+	}
+
+	defer jsonFile.Close()
+	return ParseOpenedJsonFile(jsonFile)
 }
 
 func ParseOpenedJsonFile(f *os.File) (file, error) {
-	// TODO: read the given file into your `file` object
-	return file{}, nil
+	byteValue, readError := ioutil.ReadAll(f) //ReadAll gives read data in []byte
```

Comment:
Put all comments above the code.
Please refer to [effective go](https://golang.org/doc/effective_go#commentary) when you need to know about formatting conventions. Note that these are just conventions not hard rules.

Reply:
Sure, I will read this article.